src/Plover/Reduce.hs

```diff
@@ -720,6 +744,29 @@ compileStep' ctxt (a :> b) = do
   b' <- compileStep ctxt b
   return (a' :> b')
 
+-- Control flow
+-- If -> If
+compileStep' ctxt (If (a :> c) t f) = return $ a :> (If c t f)
+compileStep' ctxt (If c t f) | compoundVal c = do
+  n <- freshName
+  return (n := c :> If (Ref n) t f)
+-- If -> If
+compileStep' ctxt (If c t f) = do
+  c' <- local $ compileStep ctxt c
+  t' <- local $ compileStep ctxt t
+  f' <- local $ compileStep ctxt f
+  return (If c' t' f')
+
+-- Vector comparison
+compileStep' _ e@(x :=: y) = do
+  tx <- local $ typeCheck x
+  ty <- local $ typeCheck y
+  case (tx, ty) of
+    -- pointwise comparison
+    (VecType (len1 : _) t1, VecType (len2 : _) t2) -> do
+      error "Vector comparison not implemented"
```

Comment:
`typeCheck` currently uses `error` everywhere, should I change them all to `left`?